src/Agent/NewRelic/Profiler/Configuration/InstrumentationConfiguration.h

```diff
@@ -172,6 +202,12 @@ namespace NewRelic { namespace Profiler { namespace Configuration
             instrumentationPoint->MethodName = GetAttributeOrEmptyString(matcherNode, _X("methodName"));
             instrumentationPoint->Parameters = TryGetAttribute(matcherNode, _X("parameters"));
 
+            // void as the parameters means parameterless method call (not all overloads)
+            if (instrumentationPoint->Parameters != nullptr && Strings::AreEqualCaseInsensitive(*(instrumentationPoint->Parameters), _X("void")))
+            {
+                instrumentationPoint->Parameters = std::unique_ptr<xstring_t>(new xstring_t());
```

Comment:
Is there a better way to handle this case so that we don't have to dynamically allocate a string? Are we missing the concept of something similar to C#'s `string.Empty`?

Reply:
I couldn't figure out a better way to do this.  But, I am no C++ master.

Reply:
I think that we might be able to use something like the following `std::make_unique<SomeType>(constructor, parameters, here);`

Reply:
I just realized that this might not be necessary at all...My modern C++ skills are not strong either. The Unique_Ptr should automatically handle that deallocation for us. So I don't think that we need to change anything here, or in the other places where we are using `new` with one of the smart pointers.